### PR TITLE
[all] introduce new module `http` and Batch support

### PR DESCRIPTION
**Pull request by udoprog**
This introduces a simple `http` module.

The protocol is documented in the README for `modules/http`.

### CODE REVIEW

**Review by gabrielgerhardsson — APPROVED** [review]
LGTM.
Disclaimer: I was a part of writing this, so would be good if someone else looked at it also.

File: api/src/main/java/com/spotify/ffwd/output/BatchedPluginSink.java [hunk 1]
```diff
@@ -37,4 +38,12 @@
...
      */
     public AsyncFuture<Void> sendMetrics(Collection<Metric> metrics);
+
+    /**
+     * Send the given collection of batches.
+     *
+     * @param batches Batches to send.
+     * @return A future that will be resolved when the batches have been sent.
+     */
+    public AsyncFuture<Void> sendBatches(Collection<Batch> batches);
```
Comment (nezdolik) [inline api/src/main/java/com/spotify/ffwd/output/BatchedPluginSink.java:48]: public keyword is redundant in interface method

File: api/src/main/java/com/spotify/ffwd/output/FlushingPluginSink.java [hunk 2]
```diff
@@ -402,17 +399,23 @@ Batch newBatch() {
     static class Batch {
```
Comment (nezdolik) [inline api/src/main/java/com/spotify/ffwd/output/FlushingPluginSink.java:399]: is it a good name and location for a class that represents collection of batches?

File: api/src/main/java/com/spotify/ffwd/output/BatchedPluginSink.java (hunk 1 above)
Comment (udoprog) [reply api/src/main/java/com/spotify/ffwd/output/BatchedPluginSink.java:48]: Yes, but it follows the convention already established in the file.

File: api/src/main/java/com/spotify/ffwd/output/FlushingPluginSink.java (hunk 2 above)
Comment (udoprog) [reply api/src/main/java/com/spotify/ffwd/output/FlushingPluginSink.java:399]: It's purely internal to this class, so name doesn't matter (much).

**Pull request merged**